pkg/download/mode/mode.go

```diff
@@ -72,12 +104,11 @@ func NewFile(m Mode, downloadURL string) (*DownloadFile, error) {
 		return parseFile(bts)
 	}
 
-	switch m {
-	case Sync, Async, Redirect, AsyncRedirect, None:
-		return &DownloadFile{Mode: m, DownloadURL: downloadURL}, nil
-	default:
-		return nil, errors.E(op, errors.KindBadRequest, fmt.Sprintf(invalidModeErr, m))
+	retFile := &DownloadFile{Mode: m, DownloadURL: downloadURL}
+	if err := retFile.Validate(); err != nil {
```

Comment:
Correct me if I'm wrong but: 

The old code that was removed here no longer works because in the old code we are checking the top level Mode, but in this validate function only the internal `d.Paths[n].Mode` is checked.

However, since top level Mode is now already checked above, then this Validate function is not necessary because `retFile` has no `Path`s to begin with.

Reply:
@marwan-at-work you're right. I left a few things out of this `Validate` and some tangential functions

- `DownloadFile.Validate` needs to check `DownloadURL` for well-formedness
- ^^ also needs to call `DownloadPath.Validate` for each `d.Paths[n]` - that does more than just validate its mode
- `Mode.Validate` does not check for `custom:` and `file:` values

I've made changes to address all this

I've fixed all of that